pkg/pkg.bzl

```diff
@@ -31,15 +33,32 @@ def _quote(filename, protect = "="):
     """Quote the filename, by escaping = by \\= and \\ by \\\\"""
     return filename.replace("\\", "\\\\").replace(protect, "\\" + protect)
 
+
 def _pkg_tar_impl(ctx):
     """Implementation of the pkg_tar rule."""
 
     # Files needed by rule implementation at runtime
     files = []
 
+    if ctx.attr.package_file_name:
+        output_name = ctx.attr.package_file_name
+        package_variables = ctx.attr.package_variables[PackageVariablesInfo]
+        if package_variables:
+            output_name = output_name.format(**package_variables.values)
```

Comment:
Yes. We have another issue to settle first.  Do we continue to have docs out of band or generate them from the .bzl files?  @nacl seems to be adding to the .bzl files. That is good, but we are not updating the README.me
The existing code is way under-documented in the .bzl.  
Sigh.

Reply:
Normally, I'd prefer the in-source approach and generate them a la `javadoc`.  Additional pages could be added to link between various pages and demonstrate concepts.

I'm not sure that stardoc, as it is written, is the best tool for this.  It, last I checked, prefers a [tabular format](https://github.com/bazelbuild/stardoc/blob/master/docs/stardoc_rule.md) that IMO doesn't scale well when the documentation becomes sufficiently large.

In the past, I've liked how [javadoc](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/String.html), but that sort of thing may be too complicated for `stardoc` for now.  There may already be some ways to do this, but I honestly haven't looked.

Given that rule specifications and properly-formatted macro documentation are already pretty easy to read, I think the following makes sense:

- Provide conceptual and basic rule argument documentation in external files
- Provide detailed documentation in-source

This seems to flow well in my head, but it could be complicated to manage.  I think I could be convinced either way.  Out of curiosity, what documentation practices are used internally for Blaze?

Reply:
Well... stardoc is due for a rewrite.  The Starlark TL does not like that the doc generation is tied to reflection and the bazel execution engine. And we do this so that we can generate unified docs in the face of different native code options for Bazel and Blaze. This is over-engineering. It should be able to be done statically with a parser.

The suggestion about basic docs and details in source sound good. To that, I will add that we would benefit from an examples tree of working code to demonstrate typical use cases.